src/tools/r2rdump/R2RMethod.cs

```diff
@@ -172,7 +177,7 @@ public enum GenericElementTypes
         /// <summary>
         /// Extracts the method signature from the metadata by rid
         /// </summary>
-        public R2RMethod(byte[] image, MetadataReader mdReader, uint rid, int entryPointId, GenericElementTypes[] instanceArgs, uint[] tok)
+        public R2RMethod(R2RReader r2r, MetadataReader mdReader, uint rid, int entryPointId, GenericElementTypes[] instanceArgs, uint[] tok)
```

Comment:
I'm not a big fan of this change because now you're requiring a R2RReader to be passed in to the constructor. It seems like the reason you're doing this is so you can call the GetTypeDefFullName method you described which is a refactoring to make it easier to get a type's full name.

I think refactoring that operation into its own method is a great idea, but maybe it doesn't belong in the R2RReader (especially as an instance method). Perhaps you could make that a static method that takes in two arguments: a MetadataReader object and a TypeDefinitionHandle object. Those are the only two objects the method operates on.